app/controllers/homepage_controller.rb

```diff
@@ -54,7 +54,12 @@ def index
         raise ArgumentError.new("Unknown view_type #{@view_type}")
       end
 
-    search_result = find_listings(params, per_page, compact_filter_params, includes.to_set)
+    search_param_digest = Digest::MD5.hexdigest([params, per_page, compact_filter_params, includes, request.xhr?].join)
+    view_param_digest = Digest::MD5.hexdigest([params, per_page, compact_filter_params, includes, request.xhr?, I18n.locale].join)
+
+    search_result = Rails.cache.fetch(["index_listings", @current_community.cache_key, search_param_digest]) do
```

Comment:
I think this part could be done inside the SphinxAdapter (inside the ListingIndexService).